HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,706 @@
+
+If the receiver’s account is **not** associated with the token, and they have **no** free auto-association slots, then the airdrop transfer neither succeeds nor fails, but is held in state as a *pending transfer*.
+
+### Pending Transfers
+
+When a `TokenAirdropTransaction` is handled for a receiver that has no free auto-association slots, a *pending transfer* is created and held in state. Tokens in the pending transfer are not deducted from the sender, nor are they credited to the receiver, until the receiver *claims* the airdrop. If the receiver never claims the airdrop, then the sender continues to pay for it forever, or until their account expires, is deleted, or the sender *cancels* the pending transfer.
+
+When the sender’s account is renewed, each pending transfer owned by the sender is taken into account and paid for as part of the renewal process. If the sender’s account expires and cannot be renewed, then all pending transfers for that sender are canceled. Any attempt to claim a missing or canceled pending transfer, or a pending transfer from an expired or deleted sender account, will fail. Conceptually, each pending transaction is immediately canceled at the time the account expired.
+
+All pending transfers sent by an account must be canceled before the account can be deleted.
+
+At the time a token transfer is stored as a *pending transfer*, any custom fees are assessed and stored. If a token A has custom fees of 1 HBAR, and the sender sends that token, the fee of 1 HBAR is assessed at the time of the TokenAirdrop transaction. Even if the custom fee is changed to 100 HBAR between the time the transaction is stored and claimed, no additional custom fee will be assessed to the sender.
+
+When a receiver claims a token, the claim transaction **must** be signed by the receiver, in addition to the transaction payer (who may be the same as the receiver). In essence, if an account has no free auto-association slots, then it behaves as though receiver-sig-required were true *for airdrops*. That is, the receiver must sign the transaction to receive the airdrop.
+
+A sender may cancel pending transfers for a low nominal fee. A receiver can also choose to just ignore pending transfers. It is expected that wallets may build special user workflows into their apps to allow customers to ignore pending transfers without having to actually issue a transaction to the network to cancel them. Since pending transfers are paid for by the sender, the receiver may choose not to simply filter them out in their wallet software.
```

Comment:
Good point and one the working group discussed a bit.
In this version only senders can edit their pending transfer intention.
Since there's no actual transfer potential receivers are not impacted by the existence of a pending transfer

Reply:
then the part that reads "without having to actually issue a transaction to the network to cancel them" is a bit odd to me ... seems to be referring to some other potential proposal where the user could do that or would need to do that.